queries.sql

```diff
@@ -9,3 +9,56 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals
+SELECT COUNT(*) FROM ANIMALS;
+ROLLBACK;
+SELECT COUNT(*) FROM ANIMALS;
+
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO SP1;
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) as escape_attempts FROM animals
+ GROUP BY neutered ORDER BY MAX(escape_attempts) DESC;
+SELECT neutered, MAX(weight_kg) as max_weight , MIN(weight_kg) as min_weight FROM animals
+ GROUP BY neutered;
+SELECT species, MAX(weight_kg) as max_weight , MIN(weight_kg) as min_weight FROM animals
+ GROUP BY species;
+SELECT species, AVG(escape_attempts) as avg_escape_attempts FROM animals
+WHERE date_of_birth >= '1990-01-01' AND  date_of_birth < '2000-01-01' GROUP BY species;
+SELECT neutered, AVG(escape_attempts) as avg_escape_attempts FROM animals
+ WHERE date_of_birth >= '1990-01-01' AND  date_of_birth < '2000-01-01' GROUP BY neutered;
```

Comment:
![image](https://github.com/dev-mamun/vet-clinic/assets/93039359/eecc1f15-8052-4ea8-8906-636ae91dc458)

- [ ] Your query is well constructed I guess you misunderstood the requirement for this query because we want those between the date of birth of `1990/01/01` to `2000/31/12` because when we are given the years range then it means that the years which is given later is also fully considered in the range 🤔 Kindly fix your query to get the expected result 👍